File: data/mods/java/lib/Robot.java
import java.io.InputStreamReader;
import java.io.BufferedReader;

public class Robot {

  BufferedReader r;
  public enum Tile {
      laby_name_Void, laby_name_Wall, laby_name_Rock,
      laby_name_Web, laby_name_Exit, laby_name_Unknown
  }

  public Tile laby_name_Void = Tile.laby_name_Void;
  public Tile laby_name_Wall = Tile.laby_name_Wall;
  public Tile laby_name_Rock = Tile.laby_name_Rock;
  public Tile laby_name_Web = Tile.laby_name_Web;
  public Tile laby_name_Exit = Tile.laby_name_Exit;
  public Tile laby_name_Unknown = Tile.laby_name_Unknown;

  Robot() {
      r = new BufferedReader(new InputStreamReader(System.in));
      output("start"); input();
  }

  public void output(String s) {
      System.out.format("%s%n", s);
  }

  String input() {
      try {
	  String s = r.readLine();
	  if (s.equals("quit")) System.exit(0);
	  return s;
      }
      catch (java.io.IOException e) {
	  System.exit(1);
	  return null;
      }
  }

  public void laby_name_left() {
      output("left"); input();
  }

  public void laby_name_right() {
      output("right"); input();
  }

  public void laby_name_forward() {
      output("forward"); input();
  }

  public void laby_name_take() {
      output("take"); input();
  }

  public void laby_name_drop() {
      output("drop"); input();
  }

  public void laby_name_escape() {
      output("escape"); input();
  }

  public Tile laby_name_look() {
    output("look");
    String ans = input ();
    if (ans.equals("void")) return Tile.laby_name_Void;
    if (ans.equals("wall")) return Tile.laby_name_Wall;
    if (ans.equals("rock")) return Tile.laby_name_Rock;
    if (ans.equals("web")) return Tile.laby_name_Web;
    if (ans.equals("exit")) return Tile.laby_name_Exit;
    return Tile.laby_name_Unknown;
  }

  public void laby_name_say(String s) {
      System.out.format("say %s%n", s); input();
  }
}

File: data/mods/js/lib/robot.js
importPackage(java.io);
importPackage(java.lang);

const stdin = new BufferedReader( new InputStreamReader(System['in']) )

function output(s)
{
  print(s);
}

function input()
{
  var s = stdin.readLine();
  return s;
}

function laby_name_left()
{
  output("left"); var s = input();
}
function laby_name_right()
{
  output("right"); var s = input();
}
function laby_name_forward()
{
  output("forward"); var s = input();
}
function laby_name_take()
{
  output("take"); var s = input();
}
function laby_name_drop()
{
  output("drop"); var s = input();
}
function laby_name_escape()
{
  output("escape"); var s = input();
}
function laby_name_say(s)
{
  output("say "+s); var s = input();
}



const laby_name_Void = 0;
const laby_name_Wall = 1;
const laby_name_Rock = 2;
const laby_name_Web = 3;
const laby_name_Exit = 4;
const laby_name_Unknown = 5;

function laby_name_look()
{
  output("look");
  var ans = input();
    if (ans.match("void"))
{ return laby_name_Void;}
    if (ans.match("wall"))
{ return laby_name_Wall;}
    if (ans.match("rock"))
{ return laby_name_Rock;}
    if (ans.match("web"))
{ return laby_name_Web;}
    if (ans.match("exit"))
{ return laby_name_Exit;}
    return laby_name_Unknown;
}

output("start");
var s = input();
